Core/clim-basic/extended-streams/text-formatting.lisp

```diff
@@ -93,9 +95,10 @@
     (new-margins (stream standard-page-layout)
      &aux (old-margins (stream-text-margins stream)))
   (macrolet ((thunk (edge)
-               `(unless (getf new-margins ,edge)
-                  (setf (getf new-margins ,edge)
-                        (getf old-margins ,edge)))))
+               `(let ((spec (getf new-margins ,edge)))
+                  (typecase spec
+                    (null (setf (getf new-margins ,edge) (getf old-margins ,edge)))
```

Comment:
Pull the common `setf (getf new-margins ,edge)` part out of the `typecase`.

Reply:
it doesn't change the value when spec is a list which is not empty. pulling it out would be incorrect.

Reply:
I thought with a `normalize-margin-spec` function (mentioned in another comment below) which would be the identity on (well-formed) conses, this could be `(setf (getf text-margins ,edge) (normalize-margin-spec (getf text-margins ,edge)))`.

Now that I think about it some more, why not make a function that normalizes (and checks?) a whole "margin spec plist"? It could be used here as well as in `(setf stream-text-margins) :around` and also tested.

Reply:
ditto

Reply:
done